File: src/main/java/wraith/fabricaeexnihilo/modules/fluids/BrineFluid.java
package wraith.fabricaeexnihilo.modules.fluids;

import net.minecraft.block.FluidBlock;
import net.minecraft.fluid.Fluid;
import net.minecraft.item.BucketItem;
import wraith.fabricaeexnihilo.modules.ModFluids;
import wraith.fabricaeexnihilo.modules.base.AbstractFluid;
import wraith.fabricaeexnihilo.modules.base.FluidSettings;

public class BrineFluid extends AbstractFluid {
    private static final FluidSettings FLUID_SETTINGS = new FluidSettings("brine", 0xACC4B9, false, true);
    public static final BrineFluid STILL = new BrineFluid(true);
    public static final BucketItem BUCKET = new BucketItem(STILL, ModFluids.getBucketItemSettings());
    public static final BrineFluid FLOWING = new BrineFluid(false);
    public static final FluidBlock BLOCK = new FluidBlock(STILL, ModFluids.getBlockSettings());

    @Override
    public boolean matchesType(Fluid fluid) {
        return fluid == STILL || fluid == FLOWING;
    }

    public BrineFluid(boolean isStill) {
        super(isStill, FLUID_SETTINGS,
                () -> BLOCK,
                () -> BUCKET,
                () -> FLOWING,
                () -> STILL
        );
    }

}

File: src/main/kotlin/exnihilofabrico/modules/fluids/BrineFluid.kt
package exnihilofabrico.modules.fluids

import exnihilofabrico.modules.ModFluids
import exnihilofabrico.modules.base.AbstractFluid
import exnihilofabrico.modules.base.BaseFluidBlock
import exnihilofabrico.modules.base.FluidSettings
import net.fabricmc.fabric.api.tag.TagRegistry
import net.minecraft.fluid.Fluid
import net.minecraft.item.BucketItem
import net.minecraft.util.Identifier
import java.util.function.Supplier

class BrineFluid(isStill: Boolean): AbstractFluid(isStill, fluidSettings, Supplier { block }, Supplier { bucket }, Supplier { flowing }, Supplier { still }) {
    override fun matchesType(fluid: Fluid) = fluid == still || fluid == flowing
    companion object {
        val fluidSettings = FluidSettings("brine")
        val still = BrineFluid(true)
        val flowing = BrineFluid(false)
        val bucket = BucketItem(still, ModFluids.bucketItemSettings)
        val block = BaseFluidBlock(still, ModFluids.blockSettings)
        val tag = TagRegistry.fluid(Identifier("c:brine"))
    }
}